### ISSUE TITLE
What to do when CATMAID has *bad* version string

### ISSUE THREAD

**Issue by jefferis** [jefferis/rcatmaid#103]
e.g. `2018.07.19-dev-1ad1035b96` does not parse according to the normal git strings.

**Comment by tomka** [comment]
Hi Greg, I changed the version representation in CATMAID to a new format to be more compatible with setups were CATMAID is deployed without its Git repository. This is for instance the case for Docker images that are used by VFB. I wasn't aware of you parsing this information, otherwise I would have let you know, sorry. The new format is basically `<base-version>[-dev]-<commit>`. The `<base-version>` is baked into the code base with every new release. For a regular setup based on our master branch, no `-dev` part will be present, for `dev` branch based setups it will. The commit ID length is currently limited to 10 characters. I should make this information more prominent in the CATMAID changelog.

**Comment by jefferis** [comment]
Thanks @tomka. But how do I know where we are between releases? Previously I have actually been using the commit counter part (which tells you how many commits have been made since the last release). Now that you have removed this, I don’t see how to follow multiple API changes between releases.

**Comment by tomka** [comment]
Thanks for the explanation, I wasn't aware of your use of this field. I can surely add it back in so that it becomes `<base-version>[-dev]-<commit-counter>-<commit>`.

**Comment by jefferis** [comment]
What was the reason to change the format? Not knowing all the ins and outs, I think would have left it as before except perhaps that the released versions would be date and nothing else.

**Comment by tomka** [comment]
There was no real reason to change the format. Instead of using `git describe`, the version string is now constructed manually (because we don't want to depend on the availability of a Git tag). In the process of doing that I thought it would be nice to include whether it is a stable version and include `-dev` if not. This information is however already implicitly in the commit ID of course and I agree that the commit counter is something useful. Since you make use of it and it would cause less friction in general, I will make the version string `git describe` compatible again (include the counter,  replace `dev` and add `g` again in front of the hash).

**Comment by jefferis** [comment]
That would be great! thanks.

**Comment by tomka** [comment]
The next release will go back to he `git describe` based version format we used before. I found a way to inject this `git describe` string also into our Docker images during their build. Not being able to do that before was my main motivation to change the format in the first place. This works now with the old format as well, which should make your live easier. I plan to publish a maintenance release of CATMAID closer to new year's.
